### ISSUE TITLE
Group block: child group block has multiple inner container div

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#34404]
### Description

The front-end side will output a container ( `.wp-block-group__inner-container` ) if the HTML element of the group block is the default `div` .
When group blocks are nested and the HTML element of the parent block is set to something other than `div`, the descendant group blocks will output multiple containers.

### Step-by-step reproduction instructions

1. Inserts a nested group block.
2. Select something other than a `<div>` from "HTML Element" in Advanced panel for the parent group block.

## Block List View Example

```
- Group Block ( main tag)
	- Group Block ( main tag)
		- Group Block ( div tag)
```

## HTML Outpul Result

```
<main class="wp-block-group">
	<main class="wp-block-group">
		<div class="wp-block-group">
			<div class="wp-block-group__inner-container">
				<div class="wp-block-group__inner-container">
					<div class="wp-block-group__inner-container"></div>
				</div>
			</div>
		</div>
	</main>
</main>
```

### Pre-checks

- [X] I have searched the existing issues.
- [X] I have tested with all plugins deactivated except Gutenberg.

**Comment by t-hamano** [comment]
I suspect that the cause is that the closing tag in the `div.wp-block-group__inner-container` box of the Group block is being output in the wrong position.

Here's another example.

## Step-by-step reproduction instructions
- Inserts a group block.
- Inserts two group blocks in parallel into the parent group block.
- Sets the HTML element of the parent group block to something other than a `div` tag.

## Editor View

![capture_1](https://user-images.githubusercontent.com/54422211/131540277-5b19cb09-a11f-49f2-9c8c-9f4c46b60896.png)

## Frontend View

![capture_2](https://user-images.githubusercontent.com/54422211/131540408-42eb6e6b-104d-447f-af3b-c24a21e5aef7.png)

**Comment by talldan** [comment]
This sounds like a similar issue to the one reported here - #34199

**Comment by talldan** [comment]
I could reproduce the second set of steps. An important caveat is that I think reproducing requires a classic theme. I used Twenty Twenty One. In block based themes the group block omits the inner container.

Editor:
<img width="641" alt="Screenshot 2021-10-26 at 1 58 00 pm" src="https://user-images.githubusercontent.com/677833/138817475-00b69345-a39d-4def-8c1a-f1c3ab4731b8.png">

Post:
<img width="637" alt="Screenshot 2021-10-26 at 1 58 05 pm" src="https://user-images.githubusercontent.com/677833/138817506-3c7abb91-fde6-4b70-a2b3-27429da3bf3d.png">

**Comment by davecpage** [comment]
@talldan you were correct, the issue was similar to the one on #34199 in that the regexes needed to be adjusted.

**Comment by davecpage** [comment]
@noisysocks Similar to #34199 would this also need to be ported over to the WordPress core for inclusion in 5.9?

**Comment by noisysocks** [comment]
The fix https://github.com/WordPress/gutenberg/pull/36256 landed in Gutenberg 11.9 which means it's included in WP 5.9.
